libr/core/disasm.c

```diff
@@ -353,6 +354,59 @@ static RAnalFunction *fcnIn(RDisasmState *ds, ut64 at, int type) {
 	return r_anal_get_fcn_in (ds->core->anal, at, type);
 }
 
+static const char *get_utf8_char (const char line, RDisasmState *ds) {
+	switch (line) {
+		case '<': return ds->core->cons->vline[ARROW_LEFT];
+		case '>': return ds->core->cons->vline[ARROW_RIGHT];
+		case ':': return ds->core->cons->vline[LINE_UP];
+		case '|': return ds->core->cons->vline[LINE_VERT];
+		case '=': return ds->core->cons->vline[LINE_HORIZ];
+		case '-': return ds->core->cons->vline[LINE_HORIZ];
+		case ',': return ds->core->cons->vline[CORNER_TL];
+		case '.': return ds->core->cons->vline[CORNER_TR];
+		case '`': return ds->core->cons->vline[CORNER_BL];
+		default: return " ";
+	}
+}
+static void ds_print_ref_lines(char *line, char *line_col, RDisasmState *ds) {
+	int i;
+	if (ds->core->cons->use_utf8 || ds->linesopts & R_ANAL_REFLINE_TYPE_UTF8) {
+		if (ds->show_color) {
+			for (i = 0; i < strlen (line); i++) {
+				if (line[i] == ' ') {
+					r_cons_printf (" ");
+					continue;
+				}
+				if (line_col[i] == 'd') {
+					r_cons_printf ("%s%s%s", COLOR (ds, color_flow), get_utf8_char (line[i], ds), COLOR_RESET (ds));
+				} else	{
+					r_cons_printf ("%s%s%s", COLOR (ds, color_flow2), get_utf8_char (line[i], ds), COLOR_RESET (ds));
+				}
+			}
+		} else {
+			for (i = 0; i < strlen (line); ++i) {
```

Comment:
the leength of line doesnt changes in the loop. so move the strlen one line above